File: src/main/webapp/resources/assets/js/app/services/equipoService.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

var servicesModule = angular.module('saircoAppServices');
servicesModule.factory('equipoService', ['$http', function ($http) {
        return {
            apiUrl: apiUrl,
            getAllEquipos: function () {
                return $http.get(this.apiUrl + 'equipo/');
            },
            getEquipoById: function (equipoId) {
                return $http.get(this.apiUrl + 'equipo/' + equipoId);
            },
            getBySalonId: function (SalonByEquipoId) {
                return $http.get(this.apiUrl + 'salon/' + salonId);
            },
            createEquipo: function (equipo) {
                return $http.put(this.apiUrl + 'equipo/', equipo);
            },
            addSalonToEquipo: function (salon,  equipoId) {
                return $http.post(this.apiUrl + 'equipo/'+equipoId+'/salon', salon);
            },
            saveOrUpdateEquipo: function (equipo) {
                return $http.post(this.apiUrl + 'equipo/'+equipo.id+equipo);
            }
        };
    }]);

File: src/main/java/co/utb/softeng/moviesapp/services/EquipoService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package co.utb.softeng.moviesapp.services;

import co.utb.softeng.moviesapp.entities.Equipo;
import co.utb.softeng.moviesapp.entities.Salon;
import java.util.List;

/**
 *
 * @author Juan Mantilla
 */
public interface EquipoService {
    
    public List<Equipo> getAllEquipos();
    public Equipo getEquipoById(Long id);
    public List<Equipo> getEquipoByName(String name);
    public List<Equipo> getEquipoBySalonId(Long salonId);
    public List<Equipo> getEquipoBySalonName(String salonName);
    public void saveOrUpdateEquipo(Equipo equipo);   
    public Equipo addSalonToEquipo(String salon, Long equipoId);
    public void deleteEquipo(Long id);
    
}